### ISSUE TITLE
Tests failing with the error "This method should only be called by Robolectric when running all tests"

### ISSUE THREAD

**Issue by Murtaza0xFF** [robolectric/robolectric#3869]
The tests pass on AS, but they fail with the following error when those exact same tests are run on CircleCI.

```
java.lang.IllegalStateException: this method should only be called by Robolectric
at org.robolectric.shadows.ShadowDisplayManager.configureDefaultDisplay(ShadowDisplayManager.java:43)
at org.robolectric.android.Bootstrap.setUpDisplay(Bootstrap.java:19)
at org.robolectric.android.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:159)
at org.robolectric.RobolectricTestRunner.beforeTest(RobolectricTestRunner.java:335)
at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:245)
at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:114)
at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:57)
at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:66)
at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
at sun.reflect.GeneratedMethodAccessor35.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
at com.sun.proxy.$Proxy1.processTestClass(Unknown Source)
at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:108)
at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:146)
at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:128)
at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
at java.lang.Thread.run(Thread.java:745)
```
#3836 reports a similar issue, but that issue was closed before it was resolved.

Using Robolectric 3.8, with targetSDKVersion and compileSDKVersion set to 27.

**Comment by BruninhoMattoso** [comment]
I get the same issue here

**Comment by sarmadali20** [comment]
getting the same one of my colleague has his tests passing this seems to be becoming it's working on my machine! not sure what I am missing!

**Comment by GeBeater** [comment]
Same here! :-(

**Comment by bornest** [comment]
For me it was a conflict with Espresso: after I removed Espresso dependency, tests started working again.

**Comment by jongerrish** [comment]
The method is being called twice, so I suspect you have an earlier error in a previous test that is leaving the state its checking as initialized. Check for previous errors

**Comment by mtotschnig** [comment]
I can confirm @jongerrish suspicion. I see this same error as a consequence of a different test that runs earlier and fails.

**Comment by hauthorn** [comment]
@jongerrish Why does this give different results when run through AS and through the gradle task? I think I have the same issue as the original poster.

**Comment by GeBeater** [comment]
@jongerrish you are right! There was a previous error. Many thanks.

**Comment by eschlenz** [comment]
I posted a solution that worked for me here: https://github.com/robolectric/robolectric/issues/3836

TL;DR is that I commented out this line in our `TestApp` (extends App [Application]), and everything started working again:

```
GoogleAnalytics.getInstance(this).setAppOptOut(true);
```

**Comment by PayneJay** [comment]
I tried to Using Robolectric 3.8, with targetSDKVersion and compileSDKVersion set to 27, but also without success. And my TestApp doesn't have `GoogleAnalytics.getInstance(this).setAppOptOut(true);`,I run with AS is OK, but run with command"./gradlew -D org.gradle.jvmargs="-noverify" test
" on Terminal will fail with this error...

**Comment by hauthorn** [comment]
@jongerrish That's correct, I get the error:

> java.lang.VerifyError: class org.robolectric.android.fakes.RoboMonitoringInstrumentation overrides final method specifyDexMakerCacheProperty.()V

And the following just fails.

Which is mentioned in #3839. I still don't understand how I am able to run my unit-tests in Android Studio just fine, but ./gradlew test (or testDebugUnitTest) fails.

**Comment by PayneJay** [comment]
@hauthorn Maybe this issue can help you [https://github.com/robolectric/robolectric-gradle-plugin/issues/144](url)

**Comment by PayneJay** [comment]
@jongerrish I find the first error,It's `java.lang.VerifyError: Expecting a stackmap frame at branch target 17`,And I got the answer at this issue [https://github.com/robolectric/robolectric-gradle-plugin/issues/144](url)

**Comment by hauthorn** [comment]
@PayneJay 

I have 

    testOptions {
        unitTests {
            includeAndroidResources = true

            all {
                jvmArgs '-noverify'
            }
        }
    }

Inside the apps build.gradle (in the android block). This doesn't seem to help. Is there something else I should try?

**Comment by PayneJay** [comment]
@hauthorn Try to add this in your dependencies `testImplementation "com.android.support.test:monitor:1.0.2"
`

**Comment by hauthorn** [comment]
I have these already (including that one)

    testImplementation 'junit:junit:4.12'
    testImplementation 'org.robolectric:robolectric:3.8'
    testImplementation 'org.robolectric:shadows-support-v4:3.3.2'
    testImplementation 'com.android.support.test:monitor:1.0.2'
    testImplementation 'org.mockito:mockito-core:2.18.3'
    testImplementation 'com.mateuszkoslacz.moviper:moviper-test:2.0.4'

I should mention that i use the kotlin plugins.

**Comment by PayneJay** [comment]
@hauthorn That is more than I can tell...[Spread out]

**Comment by kenyee** [comment]
Ditto...I see similar behavior.  First failure is the real cause.  Ignore the rest of the errors.
Hopefully RoboElectric 4.0 fixes this?

**Comment by jongerrish** [comment]
I'm going to close this issue.

This is a issue that manifests when a previous test has caused an error during initialization.

Anyone who is facing this, please find the previous failure and open an issue specific to that failure and we'll take a look

**Comment by JabezNzomo99** [comment]
##Mark
